com.microsoft.mrtk.uxcore/Dialog/Dialog.cs

```diff
@@ -2,259 +2,257 @@
-            if (destroyDialog)
-            {
-                Destroy(gameObject);
-            }
-            else
-            {
-                gameObject.SetActive(false);
-            }
+            header = null;
+            body = null;
+            positiveAction?.RemoveAllListeners();
+            positiveAction = null;
+            negativeAction?.RemoveAllListeners();
+            negativeAction = null;
+            neutralAction?.RemoveAllListeners();
+            neutralAction = null;
+
+            hasDismissed = false;
         }
 
         /// <summary>
-        /// Opens a dialog
+        /// Adds the listeners/actions to the buttons that have been
+        /// specified/added to the dialog.
         /// </summary>
-        /// <param name="property">DialogProperty class object which contains information such as title and description text</param>
-        /// <param name="placeForNearInteraction">Use placement optimized for near interaction instead of far interaction</param>
-        public virtual void Open(DialogProperty property = null, bool? placeForNearInteraction = null)
+        protected virtual void Awake()
         {
-            State = DialogState.Opening;
-            if (!gameObject.activeSelf)
+            if (negativeButton.Interactable != null)
             {
-                gameObject.SetActive(true);
+                negativeButton.Interactable.OnClicked.AddListener( () => {
+                    negativeAction.Invoke(new DialogButtonEventArgs() {
+                        ButtonType = DialogButtonType.Negative,
+                        ButtonText = negativeButton.Label.text,
+                        Dialog = this
+                    });
+                    Dismiss();
+                });
             }
-            if (property != null)
+
+            if (positiveButton.Interactable != null)
             {
-                Property = property;
-                Property.TargetDialog = this;
+                positiveButton.Interactable.OnClicked.AddListener( () => {
+                    positiveAction.Invoke(new DialogButtonEventArgs() {
+                        ButtonType = DialogButtonType.Positive,
+                        ButtonText = positiveButton.Label.text,
+                        Dialog = this
+                    });
+                    Dismiss();
+                });
             }
-            else if (Property == null)
+
+            if (neutralButton.Interactable != null)
             {
-                Debug.LogError("Cannot open the dialog because the property is not specified");
-                return;
+                neutralButton.Interactable.OnClicked.AddListener( () => {
+                    neutralAction.Invoke(new DialogButtonEventArgs() {
+                        ButtonType = DialogButtonType.Neutral,
+                        ButtonText = neutralButton.Label.text,
+                        Dialog = this
+                    });
+                    Dismiss();
```

Comment:
I think once a dialog has an optional dismissal, we start needing other features on this component to ensure we can eventually clear it (i.e. a window dismissal x icon). I think it's fine to forgoe the dismissal for this version, but it's definitely a real consideration

![image](https://user-images.githubusercontent.com/39840334/205136882-a375958d-b96c-406b-a901-d436f2414050.png)

Reply:
Talking with Max, some sort of optional "learn more" button would _not_ be part of the core options/dialog API because it wouldn't fundamentally be a "choice", it would just be some extra content you'd have on your prefab or template.